godot-core/src/registry/class.rs

```diff
@@ -458,6 +467,26 @@ fn fill_class_info(item: PluginItem, c: &mut ClassRegistrationInfo) {
             .expect("duplicate: recreate_instance_func (def)");
 
             c.godot_params.is_exposed = sys::conv::bool_to_sys(!is_internal);
+            #[cfg(before_api = "4.4")]
+            let _ = icon; // mark used
+            #[cfg(since_api = "4.4")]
+            if let Some(icon_path) = icon {
+                // Convert to GString and store in global map to keep it alive for program lifetime
+                let icon_gstring = crate::builtin::GString::from(icon_path);
+
+                let mut icon_map = global_icon_strings();
+                icon_map.insert(c.class_name, icon_gstring);
+
+                // Get pointer after insertion, while lock is still held
+                // SAFETY: The GString is stored in a static HashMap, so the pointer remains valid
+                // even after the lock guard is dropped. We must retrieve the pointer while holding
+                // the lock to ensure the HashMap isn't being modified concurrently.
+                let icon_ptr = icon_map.get(&c.class_name).unwrap().string_sys()
+                    as sys::GDExtensionConstStringPtr;
```

Comment:
it is recommended to use the `cast` method on raw pointers instead of `ptr as ...`, so as to avoid accidentally mixing up (or introducing) mutability. I know that `as` is ultra-present in the codebase, we are moving away fro mit.

https://doc.rust-lang.org/stable/std/primitive.pointer.html#method.cast-1
https://doc.rust-lang.org/stable/std/primitive.pointer.html#method.cast